snapshot-cross-account-replicator/functions/lambda.py

```diff
@@ -55,39 +63,91 @@ def get_assumed_role_rds_client(iam_role_arn, region):
 def copy_snapshot(snapshot, rds, source_region):
     """Triggers a local copy of a snapshot using the provided RDS client"""
 
+    tags = [
+        {
+            'Key': 'created_by',
+            'Value': setup_name
+        },
+        {
+            'Key': 'replicating',
+            'Value': 'false'
+        }
+    ]
+
     try:
-        rds.copy_db_snapshot(
-            SourceDBSnapshotIdentifier=snapshot['DBSnapshotArn'],
-            TargetDBSnapshotIdentifier=snapshot['DBSnapshotIdentifier'],
-            KmsKeyId=target_account_kms_key_arn,
-            SourceRegion=source_region,
-            Tags=[
-                {
-                    'Key': 'created_by',
-                    'Value': setup_name
-                }
-            ]
-        )
+        if is_cluster:
+            rds.copy_db_cluster_snapshot(
+                SourceDBClusterSnapshotIdentifier=snapshot['DBClusterSnapshotArn'],
+                TargetDBClusterSnapshotIdentifier=snapshot['DBClusterSnapshotIdentifier'],
+                KmsKeyId=target_account_kms_key_arn,
+                SourceRegion=source_region,
+                Tags=tags
+            )
+        else:
+            rds.copy_db_snapshot(
+                SourceDBSnapshotIdentifier=snapshot['DBSnapshotArn'],
+                TargetDBSnapshotIdentifier=snapshot['DBSnapshotIdentifier'],
+                KmsKeyId=target_account_kms_key_arn,
+                SourceRegion=source_region,
+                Tags=tags
+            )
     except botocore.exceptions.ClientError as e:
         raise Exception("Could not issue copy command: %s" % e)
 
 
-def match_tags(snapshot):
-    """Checks if the snapshot was created by the current setup"""
+def match_tags(snapshot, replication_status='false'):
```

Comment:
:thinking: where are you calling this function with a `replication_status` different than `false`?

Reply:
Nowhere, it's a leftover, but doesn't harm to stay either (maybe for future use).